### PR TITLE
feat(runtime): L3 post-fork host-buffer registration

### LINKED ISSUE #1027: L3 worker cannot see host tensors created after worker startup
## Summary

`DistributedWorker` / L3 worker currently requires all host input/output tensors passed to `worker.run(...)` to be created and moved to shared memory before the L3 worker child process is created. If a later run uses a newly-created tensor, or a tensor with a shape/storage that was not shared before worker creation, the child process cannot reliably see that storage.

This creates a practical limitation for dynamic-shape serving workloads: every possible input/output buffer shape must be preallocated up front, otherwise repeated `worker.run(...)` calls with changed input shapes may fail or require awkward fixed-size buffer workarounds.

## Context

This was observed while migrating `pypto-serving` Qwen3-14B prefill/decode dispatch from the L2 worker path to a shared L3 `DistributedWorker` path:

- Feature issue: https://github.com/hw-native-sys/pypto-serving/issues/26
- Implementation PR: https://github.com/hw-native-sys/pypto-serving/pull/29

In that PR, serving currently has to preallocate fixed-size host buffers for prefill/decode before creating the L3 worker:

- prefill hidden / seq_lens / chunk_lens / chunk_offsets / block_table / slot_mapping / logits
- decode hidden / seq_lens / block_table / slot_mapping / logits

The worker is then created once and reused for both prefill and decode. This works, but only because all runtime input/output buffers have stable shared-memory storage before the child process starts.

## Problem

For dynamic serving workloads, input shapes can change between runs:

- prefill total token count changes per request batch / chunk
- user batch size changes
- block table and slot mapping sizes may vary with max sequence / page layout
- output logits buffers may vary with batch shape in other models

With the current L3 worker behavior, tensors created after the worker child exists are not visible to that child unless they were already backed by shared memory visible at fork/spawn time. This means a natural pattern like:

```python
worker = DistributedWorker([program])

for request_batch in batches:
    hidden = torch.empty((dynamic_tokens, hidden_size), dtype=torch.bfloat16)
    out = torch.empty((dynamic_batch, vocab), dtype=torch.float32)
    worker.run(program, hidden, out)
```

is not safe unless `hidden` and `out` are replaced with slices of pre-created shared-memory buffers.

## Current workaround

`pypto-serving` now preallocates maximum-capacity buffers before L3 worker creation and reuses slices for each run. For example:

- prefill uses a fixed `kernel_batch * max_seq` hidden buffer but passes only the active token slice
- fixed-size metadata buffers are zeroed / filled each run
- decode uses fixed `kernel_batch` buffers and pads inactive rows

This workaround is functional, but it has drawbacks:

- higher memory footprint because maximum-size buffers must exist up front
- model runner code becomes more complex
- dynamic-shape programs need fixed-capacity host buffers even when only a small shape is active
- shape changes are constrained by preallocated capacity
- child-process visibility details leak into serving-side code

## Expected behavior

The L3 worker runtime should support one of the following:

1. Passing newly-created shared-memory host tensors to an already-running L3 worker, with the child process able to attach/map them for that run.
2. A documented runtime API to register or refresh host tensor storage mappings after worker creation.
3. A clear error if a non-visible host tensor is passed to `worker.run(...)`, instead of failing indirectly or forcing users to discover the preallocation requirement.
4. Documentation explaining the exact lifetime/visibility rules for host tensor arguments used by `DistributedWorker.run(...)`.

## Why this matters

Serving engines naturally have dynamic request batches. Requiring every possible host input/output tensor to be created before worker startup makes the L3 worker path harder to use as a general replacement for the L2 runtime path.

The limitation is manageable for Qwen3-14B today through fixed-size buffers, but it becomes fragile for other models, variable vocab/logit outputs, dynamic prefill chunking, or future serving paths that need more flexible input shapes.

## Validation from pypto-serving

After applying the fixed preallocation workaround in `pypto-serving`, these checks pass:

- offline generation without larger PTO2 runtime envs
- single-request HTTP serving without larger PTO2 runtime envs
- multi-request HTTP serving with:

```text
PTO2_RING_HEAP=4294967296
PTO2_RING_TASK_WINDOW=1048576
PTO2_RING_DEP_POOL=1048576
```

This confirms the workaround is viable, but the underlying L3 child-process tensor visibility limitation remains.

### PR DESCRIPTION
**Pull request by doraemonmj**
Closes #1027.

## Problem

L3 chip children are forked lazily on the first `Worker.run()`. A host tensor created *after* that — the natural dynamic-shape serving pattern — is not in the children's address space: the orch fn runs in the parent and the per-task args carry a raw parent VA that is unmapped (or stale) in the child. Today serving must preallocate every input/output buffer (at max shape) before the worker is created.

## Change

Add `Worker.register_host_buffer(tensor)` / `unregister_host_buffer(handle)`:

- A separate named shared-memory buffer is mapped into **every** chip child post-fork and kept mapped (broadcast via the existing NEXT_LEVEL control path, with the same partial-failure rollback as `_CTRL_PY_REGISTER`).
- Before the runtime dereferences a task, the child rewrites the mailbox blob's host pointers (`Tensor.buffer.addr`) to its own mapping — pure-Python blob rewrite, **no runtime C++ change**.
- Each `run()` mirrors the tensor through the shm: H2D copy-in before the task, D2H copy-out after the run drains.
- `submit_next_level` validates host-tensor visibility and raises an actionable error for an unregistered post-fork tensor. Fork-inherited vs post-fork is decided by the buffer's backing inode plus a fork-time VA snapshot, so a fresh post-fork `torch.empty` (its own mmap) and a post-fork `share_memory_` tensor (new inode, even if it reused a freed VA) are both rejected.

Sub-views of a registered buffer resolve automatically; views overrunning the registered size raise.

Per the issue decision this delivers **B + C + D** and defers **A** (transparent auto-mapping, which would have to guess child visibility from a raw `data_ptr` and risk silent corruption when torch recycles the pointer).

## Scope / limits (documented in `comm-domain.md`)

- memcpy, not zero-copy (a registered buffer is a separate shm); true zero-copy is a later optimization.
- `orch.copy_to` (the explicit low-level staging path) is out of scope — its `src` must still be fork-inherited.
- A small non-shared post-fork tensor sub-sliced out of a fork-time heap arena can slip the check (anonymous, inside a fork range); always `share_memory_` or register host tensors for chip dispatch.

## Files

- `python/simpler/worker.py`, `python/simpler/orchestrator.py` — implementation (Python only)
- `docs/comm-domain.md` — lifetime/visibility contract
- `tests/st/a2a3/tensormap_and_ringbuffer/test_l3_host_buffer_registration.py` — one a2a3 scene test covering the mechanism (B) and the error path (C, both shared and anonymous post-fork memory)

## Testing

- a2a3sim and a2a3 onboard (via task-submit): the scene test passes.
- pre-commit (ruff / pyright / markdownlint / english-only / headers) clean.

### CODE REVIEW

File: python/simpler/worker.py [hunk 1]
```diff
@@ -334,6 +362,195 @@ class _RemoteSession:
...
+    shm_base: int
+    tensor: Any
+
+
+@dataclass(frozen=True)
+class HostBufferHandle:
+    """Opaque handle returned by ``Worker.register_host_buffer``."""
+
+    token: int
+    data_ptr: int
```
[the remark this answers is not in the document]
Comment (doraemonmj) [reply python/simpler/worker.py:391]: The code has been matched by token.
